cli/napi/async.rs

```diff
@@ -39,6 +39,7 @@ fn napi_cancel_async_work(
 /// Frees a previously allocated work object.
 #[napi_sym::napi_sym]
 fn napi_delete_async_work(_env: &mut Env, work: napi_async_work) -> Result {
+  #[allow(clippy::from_raw_with_void_ptr)]
```

Comment:
https://rust-lang.github.io/rust-clippy/master/index.html#from_raw_with_void_ptr

I feel like this shouldn't be allowed, but I'm not sure what to do here. cc @bartlomieju @littledivy

Reply:
` as *mut AsyncWork`?